src/Microsoft.Diagnostics.DebugServices.Implementation/RuntimeService.cs

```diff
@@ -50,14 +52,10 @@ public IEnumerable<IRuntime> EnumerateRuntimes()
             if (_runtimes is null)
             {
                 _runtimes = new List<IRuntime>();
-
-                IEnumerable<IRuntimeProvider> providers = _services.GetServices<IRuntimeProvider>();
-                if (providers is not null)
+                foreach (ServiceFactory factory in _serviceManager.EnumerateProviderFactories(typeof(IRuntimeProvider)))
                 {
-                    foreach (IRuntimeProvider provider in providers)
-                    {
-                        _runtimes.AddRange(provider.EnumerateRuntimes(_runtimes.Count));
-                    }
+                    IRuntimeProvider provider = (IRuntimeProvider)factory(_services);
```

Comment:
When running under windbg/lldb (live case), do we already flush the target? So that newly added runtimes can be considered.

Reply:
Yes, every time lldb or dbgeng stops in the debugger, everything (including all the runtimes) is flushed. 

This is actually little different from the current code which only flushed (set _runtimes to null) the runtime service if there were 0 runtimes on the list.  The new behavior may add some time to re-enumerate the runtimes but it is cleaner.